### ISSUE TITLE
Recent nightly build can not precompile

### ISSUE THREAD

**Issue by BenLauwens** [JuliaCollections/DataStructures.jl#327]
Hi

A recent nightly build of Julia v0.7 can no longer precompile `DataStructures`:
```julia
julia> VERSION
v"0.7.0-DEV.1729"

julia> using DataStructures
INFO: Precompiling module DataStructures.

WARNING: deprecated syntax "parametric method syntax OrderedDict{V}(kv::Tuple{Vararg{Pair{K, V} where K}}) around none:1".
Use "OrderedDict(kv::Tuple{Vararg{Pair{K, V} where K}}) where V" instead.

WARNING: deprecated syntax "parametric method syntax OrderedDict{V}(ps::Pair{K, V} where K...) around none:1".
Use "OrderedDict(ps::Pair{K, V} where K...) where V" instead.
ERROR: LoadError: LoadError: LoadError: type QuoteNode has no field args
Stacktrace:
 [1] @delegate(::LineNumberNode, ::Module, ::Any, ::Any) at /Users/ben/.julia/v0.7/DataStructures/src/delegate.jl:4
 [2] include_relative(::Module, ::String) at ./loading.jl:464
 [3] include at ./sysimg.jl:14 [inlined]
 [4] include(::String) at /Users/ben/.julia/v0.7/DataStructures/src/DataStructures.jl:3
 [5] include_relative(::Module, ::String) at ./loading.jl:464
 [6] include(::Module, ::String) at ./sysimg.jl:14
 [7] anonymous at ./<missing>:2
while loading /Users/ben/.julia/v0.7/DataStructures/src/ordered_set.jl, in expression starting on line 20
while loading /Users/ben/.julia/v0.7/DataStructures/src/ordered_set.jl, in expression starting on line 20
while loading /Users/ben/.julia/v0.7/DataStructures/src/DataStructures.jl, in expression starting on line 70
ERROR: Failed to precompile DataStructures to /Users/ben/.julia/lib/v0.7/DataStructures.ji.
Stacktrace:
 [1] compilecache(::String) at ./loading.jl:598
 [2] _require(::Symbol) at ./loading.jl:407
 [3] require(::Symbol) at ./loading.jl:318
```

**Comment by DrKrar** [comment]
I have the same problem,
 Version 0.7.0-DEV.1738

**Comment by kmsquire** [comment]
I'm guessing there were some recent parsing changes in v0.7. The delegate macros will have to be updated to accommodate this.  See [here](https://github.com/JuliaCollections/DataStructures.jl/blob/08683b47824957bd057d0db448d3efdfdc632cf7/src/delegate.jl).

**Comment by TotalVerb** [comment]
This is https://github.com/JuliaLang/julia/issues/23661; I would defer taking action until it's confirmed to be an intentional change in Base.

**Comment by DrKrar** [comment]
Dear @kmsquire, I agree with you.
but I can not tell how to resolve for such change.

**Comment by kmsquire** [comment]
Fixed by #329

**Comment by mauro3** [comment]
Can a new release be tagged?
